scipy/signal/meson.build

```diff
@@ -0,0 +1,144 @@
+correlate_nd_c = custom_target('correlate_nd',
+   output : 'correlate_nd.c',
+   input : 'correlate_nd.c.in',
+   command : [py3, tempita, '@INPUT@', '-o', '@OUTDIR@'],
+ )
+
+correlate_nd = py3.extension_module('correlate_nd',
+   correlate_nd_c,
+   include_directories: ['.', inc_np],
+   dependencies: [py3_dep],
+   install: true,
+   subdir: 'scipy/signal'
+)
+
+lfilter_c = custom_target('lfilter',
+   output : 'lfilter.c',
+   input : 'lfilter.c.in',
+   command : [py3, tempita, '@INPUT@', '-o', '@OUTDIR@'],
+ )
+
+lfilter = py3.extension_module('lfilter',
+   lfilter_c,
+   include_directories: ['.', inc_np],
+   dependencies: [py3_dep],
+   install: true,
+   subdir: 'scipy/signal'
+)
+
+sigtools = py3.extension_module('sigtools',
+  ['sigtoolsmodule.c', 'firfilter.c', 'sigtools.h',
+  'medianfilter.c', lfilter, correlate_nd],
```

Comment:
The problem here turned out to be a little different. Had I read the `setup.py` file more carefully I would have seen it sooner. This wasn't necessary:
```python
lfilter = py3.extension_module('lfilter',
```
There's no such extension in `setup.py`. So that could have been left out, and instead the line here should have been:
```
'medianfilter.c', lfilter_c, correlate_nd_c],  # note the `_c`
```

Reply:
Yeah, `extension_module` means shared binaries (`so` files). I believe they are modules which can be accessed from python code. Linking them with `link_with` isn't possible. That's the issue, but its great that it is fixed. Awesome.